server/odc-service/src/main/java/com/oceanbase/odc/service/session/ConnectConsoleService.java

```diff
@@ -341,6 +446,22 @@ public List<SqlExecuteResult> getAsyncResult(@NotNull String sessionId, String r
         }
     }
 
+    public AsyncExecuteResultResp getAsyncResultV2(@NotNull String sessionId, String requestId) {
+        PreConditions.validArgumentState(Objects.nonNull(requestId), ErrorCodes.SqlRegulationRuleBlocked, null, null);
+        ConnectionSession connectionSession = sessionService.nullSafeGet(sessionId);
+        AsyncExecuteContext<SqlExecuteResult> executeContext =
+                (AsyncExecuteContext) ConnectionSessionUtil.getExecuteContext(connectionSession, requestId);
+        if (executeContext.getFuture().isDone()) {
```

Comment:
we may not know the detail of the context~

Reply:
done